File: sdks/csharp/rest/src/Brume.Rest/Channels/Requests/ConnectRequest.cs
using Brume.Rest.Core;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record ConnectRequest
{
    /// <summary>
    /// Legacy JWT query parameter. Prefer the `Sec-WebSocket-Protocol: brume.token.&lt;jwt&gt;` subprotocol, which keeps the token out of access logs, browser history, and the `Referer` header.
    /// </summary>
    [JsonIgnore]
    public string? Token { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Channels/Requests/ConnectRequest.php
<?php

namespace Brume\Rest\Channels\Requests;

use Brume\Rest\Core\Json\JsonSerializableType;

class ConnectRequest extends JsonSerializableType
{
    /**
     * @var ?string $token Legacy JWT query parameter. Prefer the `Sec-WebSocket-Protocol: brume.token.<jwt>` subprotocol, which keeps the token out of access logs, browser history, and the `Referer` header.
     */
    public ?string $token;

    /**
     * @param array{
     *   token?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->token = $values['token'] ?? null;
    }
}
